src/components/Draggable/FilterDraggable.tsx

```diff
@@ -0,0 +1,219 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React, { useState } from 'react'
+import { useDrag } from 'react-dnd'
+import { FixedSizeList } from 'react-window'
+import { Button, Checkbox, Dropdown, DropdownItem, HFlow, Icon, TextField, Tooltip } from '..'
+import { useTheme } from '../../styles'
+import { DraggableProps } from './Draggable'
+
+export interface FilterDraggableProps<T> extends NonNullable<DraggableProps<T>> {
+  styles: any
+}
+
+export function FilterDraggable<T>(props: FilterDraggableProps<T>) {
+  const {
+    name,
+    type,
+    origin,
+    value,
+    filterValues,
+    filterState,
+    onDragEnd,
+    handleFilterUpdate,
+    formatter,
+    onKeyNav,
+    styles,
+  } = props
+
+  const [searchedFilterSet, setSearchedFilterSet] = useState<Array<string>>(filterValues)
+
+  const [open, setOpen] = useState(false)
+
+  const [all, setAll] = useState<0 | 1 | 2>(
+    filterState.size === 0 ? 0 : filterState.size === filterValues.length ? 2 : 1
+  )
+
+  const [buttonRef, setButtonRef] = useState<HTMLButtonElement>()
+
+  const theme = useTheme()
+
+  const [{ isDragging }, drag] = useDrag({
+    item: { type, name: name, origin },
+    end: (_item, monitor) => {
+      if (monitor.getDropResult() != null) onDragEnd()
+    },
+    collect: (monitor) => ({
+      isDragging: !!monitor.isDragging(),
+    }),
+  })
+
+  const handleClick = () => (open ? handleClose() : setOpen(true))
+
+  const handleClose = () => {
+    setOpen(false)
+    setSearchedFilterSet(filterValues)
+  }
+
+  const handleSelect = (element: string) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    filterState.has(element) ? filterState.delete(element) : filterState.add(element)
+    handleFilterUpdate(name, new Set<string>(filterState))
+    if (filterState.size === 0) {
+      setAll(0)
+    } else if (filterState.size === filterValues.length) {
+      setAll(2)
+    } else {
+      setAll(1)
+    }
```

Comment:
This is the same logic used to defined the default value of `all` state. I think the way that was written there more legible.
```suggestion
      setAll(filterState.size === 0 ? 0 : filterState.size === filterValues.length ? 2 : 1)
```